services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -0,0 +1,1023 @@
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(1L);
+        } else {
+            Query query = signalStateStopEventRepo.getQuery(intersectionID, startTime, endTime, false);
+
+            long count;
+            if (fullCount) {
+                count = signalStateStopEventRepo.getQueryFullCount(query);
+            } else {
+                count = signalStateStopEventRepo.getQueryResultCount(query);
+            }
+
+            log.debug("Found: {} SignalStateStopEvents", count);
+            return ResponseEntity.ok(count);
+        }
+    }
+
+    @RequestMapping(value = "/events/signal_state_stop/daily_counts", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<List<IDCount>> getDailySignalStateStopEventCounts(
+            @RequestParam(name = "intersection_id") Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis") Long startTime,
+            @RequestParam(name = "end_time_utc_millis") Long endTime,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(MockIDCountGenerator.getDateIDCounts());
+        } else {
+            return ResponseEntity
+                    .ok(signalStateStopEventRepo.getAggregatedDailySignalStateStopEventCounts(intersectionID, startTime,
+                            endTime));
+        }
+    }
+
+    @RequestMapping(value = "/events/time_change_details", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<List<TimeChangeDetailsEvent>> findTimeChangeDetailsEvent(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<TimeChangeDetailsEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getTimeChangeDetailsEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = timeChangeDetailsEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(timeChangeDetailsEventRepo.find(query));
```

Comment:
issue(blocking): executing query second time for logging purposes

Reply:
Extra query has been removed